PYME/cluster/rules.py

```diff
@@ -568,7 +568,8 @@ def prepare(self):
         self._next_release_start = self.start_at
         numTotalFrames = self.ds.getNumSlices()
         self.frames_outstanding=numTotalFrames - self._next_release_start
-        
+        if self.data_complete:
+            return dict(max_tasks=self.ds.getNumSlices())
```

Comment:
```suggestion
            return dict(max_tasks=numTotalFrames)
```

avoid one `listdir()` call